### PR TITLE
World cup daily podcast: treat vs container test

### PR DESCRIPTION
**Pull request by tomrf1**
## What does this change?
Tests treat vs 'container' positions for the World Cup Daily podcast on the UK network front.
Supports mobile web and desktop.

The UK front will have a container backfilled with the podcast series, and a single card. This gives us access to the episode title and url. This container will be either hidden or modified client-side for the test.

Apologies for the convoluted commits - this started out as an adaptation of the previous Politics Weekly treat test, but we switched to World Cup Daily instead (the treat is still similar).

## Screenshots
Treat mobile (below the headlines container):

Treat desktop (in headlines treats spot):
wide

Container mobile (below the headlines container):

Container desktop (below the headlines container):
wide

## What is the value of this and can you measure success?
Compare click-through

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [x] Locally
- [x] On CODE (optional)

### CODE REVIEW

**Comment by regiskuckaertz** [comment]
screen shot 2018-07-02 at 11 05 14

😵

File: static/src/inline-svgs/journalism/football-weekly-container/play-btnbig.svg [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="34" height="34" viewBox="0 0 34 34">
```
Comment (regiskuckaertz) [inline static/src/inline-svgs/journalism/football-weekly-container/play-btnbig.svg]: Can @akemitakagi or you optimise the SVGs with SVGOMG?

Comment (tomrf1) [reply static/src/inline-svgs/journalism/football-weekly-container/play-btnbig.svg]: 👍 I'll do this

File: static/src/javascripts/projects/common/modules/experiments/tests/football-weekly-treat-vs-container.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+import containerWaveMobile from 'svgs/journalism/football-weekly-container/waveform-mobile.svg';
+import containerWaveTiny from 'svgs/journalism/football-weekly-container/waveform-tiny.svg';
+import containerButton from 'svgs/journalism/football-weekly-container/play-btnbig.svg';
+
+const getHeadlineText = (el: Element): string =>
+    el.textContent.split(/[-–]/)[0];
+
+const runTreatTest = (): void => {
+    // Get the headline from podcast container and insert the test treat
+    const podcastContainer = document.querySelector(
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/experiments/tests/football-weekly-treat-vs-container.js]: `document.getElementById('world-cup-daily')` (without the leading `#`) will be much faster

File: static/src/javascripts/projects/common/modules/experiments/tests/football-weekly-treat-vs-container.js [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        '.facia-page #world-cup-daily'
+    );
+    const headlinesContainer = document.querySelector('.facia-page #headlines');
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/experiments/tests/football-weekly-treat-vs-container.js]: ditto

File: static/src/javascripts/projects/common/modules/experiments/tests/football-weekly-treat-vs-container.js [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+                    const treats = headlinesContainer.querySelector(
+                        '.treats__container'
+                    );
+                    if (treats) {
+                        treats.innerHTML = html;
+                    }
+                } else {
+                    // As a new section under headlines
+                    const html = `<section class='fc-container football-weekly-treat__container'><div class='fc-container__inner'>${newTreat}</div></section>`;
+                    headlinesContainer.insertAdjacentHTML('afterend', html);
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/experiments/tests/football-weekly-treat-vs-container.js]: Nice 👍

Comment (regiskuckaertz) [reply static/src/javascripts/projects/common/modules/experiments/tests/football-weekly-treat-vs-container.js]: DOM mutations should probably go inside a `fastdom.write` thunk for good measure

File: static/src/javascripts/projects/common/modules/experiments/tests/football-weekly-treat-vs-container.js [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+                        ),
+                        waveMobile: addClassesAndTitle(
+                            containerWaveMobile.markup,
+                            ['football-weekly-container__wave-mobile']
+                        ),
+                        waveTiny: addClassesAndTitle(containerWaveTiny.markup, [
+                            'football-weekly-container__wave-tiny',
+                        ]),
+                    });
+                    oldBody.innerHTML = newContainer;
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/experiments/tests/football-weekly-treat-vs-container.js]: Same here, `fastdom.write(() => oldBody.innerHTML = newContainer)`

File: static/src/javascripts/projects/common/modules/experiments/tests/football-weekly-treat-vs-container.js [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+                    podcastContainer.className +=
+                        'football-weekly-container__visible';
+                }
+            }
+        }
+    }
+};
+
+const trackClick = (name: string) => (complete: () => void): void => {
+    const treat = document.querySelector(`.${name}`);
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/experiments/tests/football-weekly-treat-vs-container.js]: Small optimisation:
```js
const trackClick = (name: string) => {
  const treat = document.querySelector(`.${name}`);
  return (complete: () => void): void => { ... };
}
```

**Comment by regiskuckaertz** [comment]
Very clean 👍

**Pull request merged**